### ISSUE TITLE
"Go online" message shown on maintenance mode settings page when clearing caches.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3313]
## Describe your issue or idea

I bumped into this annoying little bug while working on #2583 ...and you know me and my OCDs 😄

### Steps to reproduce (if reporting a bug)

1. Navigate to the "Maintenance mode" page (`/admin/config/development/maintenance`)
2. Set the site to maintenance mode -> warning: "The site is now in maintenance mode. Only users with the "Access site in maintenance mode" permission will be able to access the site." 👍
3. Refresh page -> warning gone 👎 ...should be persistent across page refreshes, but not part of what this issue here is trying to address (see #2583).
4. Clear caches -> warning: "The site is currently in maintenance mode. Go online." 👎 this warning should be suppressed in this page.

### Actual behavior (if reporting a bug)

The "The site is currently in maintenance mode. Go online." warning is shown:

<img width="518" alt="screen shot 2018-10-13 at 9 39 38 am" src="https://user-images.githubusercontent.com/2423362/46897025-09ed5880-cecc-11e8-8132-4bbfadc63a8b.png">

### Expected behavior (if reporting a bug)

The warning shown in step 4 above should be suppressed, as per https://github.com/backdrop/backdrop/blob/1.x/core/includes/menu.inc#L4036

```
...
      // Ensure that the maintenance mode message is displayed only once
      // (allowing for page redirects) and specifically suppress its display on
      // the batch and maintenance mode settings pages.
...
```

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2319

**Comment by klonos** [comment]
...using `dpm()` revealed that when clearing caches, `current_path()` is `admin_bar/flush-cache`, so the `!in_array(current_path(), array('batch', 'admin/config/development/maintenance'))` in https://github.com/backdrop/backdrop/blob/1.x/core/includes/menu.inc#L4038 was always true. It seems that adding `admin_bar/flush-cache` to the list of paths where the warning should be suppressed fixes the issue.

...I have checked the PR sandbox, and confirmed that the warning is not affected for the rest of the pages.

**Comment by stpaultim** [comment]
@klonos - A couple of quick thoughts.

1) I find the phrase "go online" confusing in general. It took me a while to figure out what it means and at first assumed that that message itself was the bug. But, that is probably for another issue elsewhere.
2) This messages seems to show up everywhere, so I would never have recongnized that this is a bug, if you didn't tell me. But, I understand now that it is unnecessary on the maintenance mode page. 
3) When I first arrived in the sandbox, it was already in maintenance mode and I'm quite sure than the "Go online" message was showing up on the maintenance mode page. But, then I turned off maintenance mode and turned it back and was unable to recreate that behavior. 
4) Once I turned off maintenance mode and turned it back on, the "Go online" message did not appear on the maintenance mode page even when I cleared the cache. It did seem to appear everywhere else as expected. 

In short, it seems the PR is working. But, I'm unable to explain why the "Go online" messages seemed to be appearing when I first arrived in the sandbox and I can not recreate that situation.

**Comment by klonos** [comment]
Thanks for the review @stpaultim ...

1. I do not like it either, we should change it to something that makes more sense. #2583 is the place to suggest alternatives 😉

2. Also addressed in #2583 by showing a different message while on the maintenance mode page.

3. Yes, I have set the sandbox in maintenance mode to test things 😄. There are two bugs happening here, this issue is about only one of them. I have updated the issue summary to make that clear, so please follow the steps in the issue summary to reproduce the problem on your local or another site. Then try the same in the PR sandbox to see the difference.

4. 👍

**Comment by klonos** [comment]
...and here's a perhaps better explanation: in the PR in #2583 I have made the warning shown in the maintenance page be persistent across page refreshes as well as when clearing cache. But if you clear caches, the "The site is currently in maintenance mode. Go online." message is also shown (the bug I am trying to address in this issue here). The result is to have both warnings in the maintenance page when clearing caches, when only one should be shown:

<img width="1186" alt="screen shot 2018-12-11 at 9 54 17 am" src="https://user-images.githubusercontent.com/2423362/49786075-333f2f00-fd2b-11e8-8aaf-1bef23619985.png">

The reason why I have split this up to a separate issue is because this is a (mostly) 1-2 lines code fix that has more chances to get in in a bug release, whereas we might be going back and forth with different ideas/suggestions in #2583 for many releases (I hope not 😅).

**Comment by stpaultim** [comment]
@klonos - It's weird. Here is what happened. 
1) I wend to the sandbox site
2) I navigated to the maintenance mode page
3) It was already in maintenance mode
4) I cleared the cache and the message "The site is currently in maintenance mode. Go online" appeared along with the "Every cache cleared" message. See screenshot.

I then navigated away and navigated back to this page and tried to get this to happen again and was unable to recreate it (I tried many times and was unable to recreate it). But, I have a screenshot to prove it happened the first time and this is pretty much what happened to me last time I tried to test this issue. 

For some reason, on the initial clearing of cache the message seems to appear for me, but then I can't get it to come back. 

The PR seems to be mostly working. But, I am getting this weird behavior so I am reluctant to mark RTBC. Any thoughts? Is "good enough" an option here to move things forward?

![maintenance_mode___pr_2319_backdrop_backdrop_testing_site](https://user-images.githubusercontent.com/3144571/50048578-b8ad4100-0095-11e9-9634-35cdd9cd9bac.png)

**Comment by stpaultim** [comment]
I think I found the problem, but I may need more testing. If I get to the maintenance mode page via the "go online" link, that causes a problem. You are correct, that is a different issue and I may be able to mark this RTBC and open a new issue for the other problem (for find an existing ticket). 

I got to run now, will look in on this issue again later tonight.

**Comment by stpaultim** [comment]
Looking back I see that the "go online" link that takes a user to the maintenance mode page includes "destination" information that comes into play when the user clears the cache. It seems reasonable to ignore that as a different issue and I'm not even sure it's a problem that needs to be fixed. We do have another issue looking at this link https://github.com/backdrop/backdrop-issues/issues/3418. 

Taking that out of consideration, I think this PR is RTBC.

**Comment by quicksketch** [comment]
Thanks @klonos and @stpaultim! Simple fix. Thanks for testing! Merged into 1.x and 1.11.x.
